presto-hive/src/test/java/io/prestosql/plugin/hive/AbstractTestHive.java

```diff
@@ -1049,6 +1049,7 @@ protected void doTestMismatchSchemaTable(
         try (Transaction transaction = newTransaction()) {
             ConnectorSession session = newSession();
             ConnectorMetadata metadata = transaction.getMetadata();
+            metadata.beginQuery(session);
```

Comment:
Should this be part of the previous commit?  Specifically, do the tests work in the last commit?

Reply:
I will merge last two commits, i kept it this way to make review easier.